source/distributed/fully_distributed_tria.cc

```diff
@@ -285,6 +289,71 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::copy_triangulation(
+      const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      Assert(this->dynamic_construction_data,
+             ExcMessage("No DynamicConstructionData has been provided!"));
+
+      // pointer to the triangulation for which the construction data
+      // should be created (normally it is the input triangulation but
+      // in the case of a serial triangulation we create a copy which should
+      // be used)
+      const dealii::Triangulation<dim, spacedim> *other_tria_ptr = &other_tria;
+
+      // temporary serial triangulation (since we might modifying subdomain_ids
+      // and level_subdomain_ids during partitioning); it is filled only
+      // if the source triangulation is serial
+      dealii::Triangulation<dim, spacedim> serial_tria;
```

Comment:
I'd make that a `std::unique_ptr` so that we don't allocate any memory if you don't need the object.

Reply:
can you say why you actually need the variable at all? I suspect because you partition the serial triangulation and you can't do that on the input argument?

Reply:
...since it's `const`?